support/builder/build_dep_populate.sh

```diff
@@ -0,0 +1,463 @@
+            if [[ -n ${AWS_SECRET_ACCESS_KEY:-} ]]; then
+                echo ""
+                echo "AWS Credentials configured via ENVVAR detected."
+                echo ""
+                echo "aws_access_key_id=${AWS_ACCESS_KEY_ID:-}"
+                echo "aws_secret_access_key=${AWS_SECRET_ACCESS_KEY:-}"
+                credSelect
+                credsConfigured=true
+                break
+            else
+                echo ""
+                echo "WARNING: Incomplete AWS Credentials configured via ENVVAR."
+                echo "Make sure to set AWS_ACCESS_KEY_ID && AWS_SECRET_ACCESS_KEY"
+                break
+            fi
+        fi
+    done
+
+    if  "${credsConfigured}" ; then
+        echo ""
+        echo "Credentials configured!"
+    else
+        echo ""
+        echo "WARNING: No AWS credentials detected!"
+        read -r -p "Would you like to generate them now? [y/N] " response
+        if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+            genS3Config
+        else
+            echo "Please configure your AWS credentials and re-run s3migrate."
+            echo ""
+            exit 1
+        fi
+    fi
+}
+
+welcome() {
+    echo ""
+    echo "==========================================================="
+    echo "###########################################################"
+    echo "############## Bldr Build Dep Ingest/Migrate ##############"
+    echo "###########################################################"
+    echo "==========================================================="
+    echo ""
+    echo "This tool will scrape all the hart files from an s3 or minio"
+    echo "bucket and reingest them to populate build dep and tdep metadata"
+    credsConfigured=false
+
+    setBldrUrl
+
+
+
+    if [[ -n "${HAB_AUTH_TOKEN+x}" ]]; then
+        echo ""
+        echo "We were able to discover your builder auth token as: "
+        echo "${HAB_AUTH_TOKEN}"
+        read -r -p "Would you like to use these credentials? [y/N]" response
+            if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+                echo "Setting detected credentials"
+            fi
+    else
+        echo ""
+        echo "WARNING: No Builder credentials detected!"
+        read -r -p "Would you like to generate them now? [y/N] " response
+        if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+            genBldrCreds
+        else
+            echo "Please configure your builder auth-token and re-run s3migrate."
+            echo ""
+            exit 1
+        fi
+    fi
+
+    if [ "$s3type" = "minio" ]; then
+        echo ""
+        echo "It looks like you specified an ingestion from minio!"
+
+        sgroups=("default" "dev" "prod")
+        for i in "${sgroups[@]}"; do
+            if curl -s localhost:9631/services/builder-minio/"${i}" | jq .cfg.key_id > /dev/null; then
+                access_key_id=$(curl -s localhost:9631/services/builder-minio/default | jq .cfg.key_id)
```

Comment:
Should this (and the next line) be using `"${i}"` rather than `default` in the URI?

Reply:
Some of this code is some very old stuff that i did in the s3sync script. Reading this, I think you're right. But I don't know for certain. I'll change it for my tests.